packages/flame/lib/src/components/position_component.dart

```diff
@@ -332,17 +332,45 @@ class PositionComponent extends Component {
     canvas.transform(transformMatrix.storage);
   }
 
-  /// Returns the relative position/size of this component.
-  /// Relative because it might be translated by their parents (which is not considered here).
-  Rect toRect() => topLeftPosition.toPositionedRect(scaledSize);
+  /// Returns the bounding rectangle for this component.
+  ///
+  /// The bounding rectangle is given in parent's coordinate space, and is
+  /// defined as the smallest axes-aligned rectangle that can fit this
+  /// component. The aspect ratio of the bounding rectangle may be different
+  /// from [size] if the component was scaled and/or rotated.
+  Rect toRect() {
+    final topLeft = positionOfAnchor(Anchor.topLeft);
+    final bottomRight = positionOfAnchor(Anchor.bottomRight);
+    if (angle == 0) {
+      return Rect.fromPoints(topLeft.toOffset(), bottomRight.toOffset());
+    } else {
+      final topRight = positionOfAnchor(Anchor.topRight);
+      final bottomLeft = positionOfAnchor(Anchor.bottomLeft);
+      final xs = [topLeft.x, topRight.x, bottomLeft.x, bottomRight.x]..sort();
+      final ys = [topLeft.y, topRight.y, bottomLeft.y, bottomRight.y]..sort();
+      return Rect.fromLTRB(xs.first, ys.first, xs.last, ys.last);
+    }
+  }
 
-  /// Returns the absolute position/size of this component.
-  /// Absolute because it takes any possible parent position into consideration.
-  Rect toAbsoluteRect() => absoluteTopLeftPosition.toPositionedRect(scaledSize);
+  /// The bounding rectangle of the component in global coordinate space.
+  ///
+  /// This is similar to [toRect()], except the rectangle is projected into the
+  /// outermost coordinate frame.
+  Rect toAbsoluteRect() {
+    final topLeft = absolutePositionOfAnchor(Anchor.topLeft);
+    final bottomRight = absolutePositionOfAnchor(Anchor.bottomRight);
+    if (angle == 0) {
+      return Rect.fromPoints(topLeft.toOffset(), bottomRight.toOffset());
+    } else {
+      final topRight = absolutePositionOfAnchor(Anchor.topRight);
+      final bottomLeft = absolutePositionOfAnchor(Anchor.bottomLeft);
+      final xs = [topLeft.x, topRight.x, bottomLeft.x, bottomRight.x]..sort();
+      final ys = [topLeft.y, topRight.y, bottomLeft.y, bottomRight.y]..sort();
+      return Rect.fromLTRB(xs.first, ys.first, xs.last, ys.last);
+    }
+  }
 
-  /// Mutates position and size using the provided [rect] as basis.
-  /// This is a relative rect, same definition that [toRect] use
-  /// (therefore both methods are compatible, i.e. setByRect ∘ toRect = identity).
+  // FIXME
```

Comment:
What is missing here?

Reply:
1. It is not the inverse of `toRect()`: what if the component is rotated and/or scaled?
2. Do we want to reset the angle here?
3. If the component is flipped, do we want to unflip it, or keep as-is?
4. Changing the `size` of the component is questionable: it does not affect the visible size (unless the component uses its `size` property to render itself, which is not recommended), so do we want to modify the `scale` instead?

Generally, I'm not sure what the purpose of this method is and how it is expected to be used, so don't know in which direction to take it.

Reply:
@luanpotter can you answser @st-pasha here?

Either we should not replace the original docs of the method for a FIXME. We could

 - If this is indeed wrong, put a FIXME comment with why it is broken, but leave the original dartdoc on the method
 - Fix the implementation and update the dartdoc if needed

Reply:
Ok, I restored the comment to how it was before. The concerns about its functionality could be posted as a new issue instead.

Reply:
Filed as #978